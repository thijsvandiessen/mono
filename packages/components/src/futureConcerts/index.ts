export { FutureConcerts } from './futureConcerts.jsx'
