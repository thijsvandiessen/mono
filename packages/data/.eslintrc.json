{
  "ignorePatterns": ["**/generated/*.ts", "dist/*"],
  "plugins": ["@typescript-eslint", "testing-library"],
  "extends": [
    "eslint:recommended",
    "plugin:react-hooks/recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "parser": "@typescript-eslint/parser",
  "env": {
    "jest": true
  },
  "rules": {
    "require-await": ["warn"],
    "no-unused-vars": ["warn"],
    "@typescript-eslint/no-explicit-any": "warn",
    "@typescript-eslint/consistent-type-imports": "error",
    "sort-imports": [
      "warn",
      {
        "ignoreCase": false,
        "ignoreDeclarationSort": false,
        "ignoreMemberSort": false,
        "memberSyntaxSortOrder": ["none", "all", "multiple", "single"],
        "allowSeparatedGroups": false
      }
    ]
  },
  "root": true
}
