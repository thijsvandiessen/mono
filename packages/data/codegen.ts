import type { CodegenConfig } from '@graphql-codegen/cli'

const config: CodegenConfig = {
  schema: './src/generated/schema.graphql',
  documents: ['./**/*.ts'],
  emitLegacyCommonJSImports: false,
  generates: {
    './src/generated/': {
      preset: 'client',
      config: {
        withHooks: true,
        scalars: {
          BooleanType: 'boolean',
          CustomData: 'Record<string, string>',
          Date: 'string',
          DateTime: 'string',
          FloatType: 'number',
          IntType: 'number',
          ItemId: 'string',
          JsonField: 'unknown',
          MetaTagAttributes: 'Record<string, string>',
          UploadId: 'string',
        },
        useTypeImports: true,
        verbatimModuleSyntax: true,
      },
      presetConfig: {
        persistedDocuments: true,
        gqlTagName: 'gql',
        fragmentMasking: false,
      },
    },
    './src/generated/introspection.json': {
      plugins: ['introspection'],
      config: {
        minify: false,
        schemaDescription: true,
        federation: true,
      },
    },
  },
  hooks: { afterAllFileWrite: ['prettier --write'] },
  ignoreNoDocuments: true,
}

export default config
