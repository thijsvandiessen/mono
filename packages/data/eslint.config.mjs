import { defineConfig, globalIgnores } from 'eslint/config'
import { eslintConfig } from '@mono/eslint-config'

export default defineConfig([
  globalIgnores(['src/generated/*.ts', 'dist/*', 'node_modules']),
  eslintConfig,
])
