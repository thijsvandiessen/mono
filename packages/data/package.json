{
  "name": "@mono/data",
  "version": "1.0.0",
  "description": "data",
  "type": "module",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    }
  },
  "scripts": {
    "build": "tsc",
    "codegen": "graphql-codegen-esm --config ./codegen.ts",
    "codegen:watch": "graphql-codegen-esm --require dotenv/config --config ./codegen.ts --watch",
    "dev": "tsc --watch",
    "download-schema": "rover graph fetch vandiessen-mono-graph@current > ./src/generated/schema.graphql && pnpm run lint:styles",
    "lint": "pnpm run lint:js && pnpm run lint:styles",
    "lint:js": "eslint",
    "lint:styles": "prettier . --write  --list-different",
    "test": "vitest run -c ./vitest.config.ts"
  },
  "dependencies": {
    "@graphql-typed-document-node/core": "3.2.0",
    "@mono/next-js": "workspace:*",
    "@mono/utils": "workspace:*",
    "@urql/core": "6.0.1",
    "@urql/exchange-graphcache": "8.1.0",
    "@urql/exchange-retry": "2.0.0",
    "graphql": "16.11.0",
    "react": "19.2.0"
  },
  "devDependencies": {
    "@apollo/rover": "0.36.2",
    "@commitlint/config-conventional": "20.0.0",
    "@graphql-codegen/cli": "6.0.1",
    "@graphql-codegen/client-preset": "5.1.1",
    "@graphql-codegen/introspection": "5.0.0",
    "@graphql-codegen/typescript": "5.0.2",
    "@graphql-codegen/typescript-document-nodes": "5.0.2",
    "@graphql-codegen/typescript-operations": "5.0.2",
    "@mono/eslint-config": "workspace:*",
    "@mono/typescript-config": "workspace:*",
    "@testing-library/jest-dom": "6.9.1",
    "@testing-library/react": "16.3.0",
    "@types/node": "24.9.2",
    "@types/react": "19.2.2",
    "@types/react-dom": "19.2.2",
    "@vitejs/plugin-react": "5.1.0",
    "dotenv": "17.2.3",
    "eslint": "9.38.0",
    "eslint-plugin-testing-library": "7.13.3",
    "prettier": "3.6.2",
    "turbo": "2.5.8",
    "typescript": "5.9.3",
    "vite-tsconfig-paths": "5.1.4",
    "vitest": "4.0.4"
  }
}
