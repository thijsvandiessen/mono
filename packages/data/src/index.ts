export { client } from './gqlClient.js'
export {
  getConcert,
  getConcertPage,
  getConcertPageSeo,
  getConcerts,
  getConcertsMeta,
  getConcertUrls,
  getFutureConcerts,
  getGeneralInfo,
  getLocation,
  getLocations,
  getPage,
  getPages,
  getPageSeo,
  getPagesMeta,
  getSiteInfo,
  getSiteMetadata,
} from './getters/index.js'

export type {
  Author,
  BlogPost,
  Concert,
  PageLink,
  CloudinaryAsset,
  Image,
  Location,
  Music,
  Navigation,
  NavigationItem,
  SubMenuItem,
} from './types/index.js'

export { isOfTypeCloudinaryAsset } from './types/index.js'
export {
  formatCloudinaryDocument,
  formatCloudinaryImage,
  navigationFormatter,
} from './formatters/index.js'
export * from './generated/graphql.js'
