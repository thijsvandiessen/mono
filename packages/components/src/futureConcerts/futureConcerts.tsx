import { ConcertListItem } from '../concertListItem/index.js'
import React from 'react'
import { getFutureConcerts } from '@mono/data'
export interface Props {
  skip: number
  first: number
}

export const FutureConcerts = async ({ skip, first }: Props) => {
  const { data } = await getFutureConcerts({ skip, first })

  return (
    <div>
      {data.map((concert, index) => (
        <ConcertListItem
          className="content-layout--small"
          key={concert.id}
          data={concert}
          size="large"
          isLast={data.length - 1 === index}
        />
      ))}
    </div>
  )
}
