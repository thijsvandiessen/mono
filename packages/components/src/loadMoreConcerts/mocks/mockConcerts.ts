import { type Concert } from '@mono/data'

export const mockConcert: Concert = {
  id: 'mock-id',
  title: 'title',
  url: '/url',
  image: undefined,
  locations: [],
}
