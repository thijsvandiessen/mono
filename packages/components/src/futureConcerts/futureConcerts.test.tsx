import { FutureConcerts, type Props } from './futureConcerts.js'
import { describe, expect, it, vi } from 'vitest'
import { render, screen } from '@testing-library/react'
import type { Props as ConcertListItemProps } from '../concertListItem/index.js'
import { getFutureConcerts } from '@mono/data'
import { mockConcert } from './mocks/mockConcerts.js'
import { resolvedComponent } from '@mono/utils'

vi.mock('@mono/data', () => {
  return {
    __esModule: true,
    getFutureConcerts: vi.fn(),
  }
})

vi.mock('../concertListItem/index.js', () => {
  return {
    __esModule: true,
    ConcertListItem: ({ data }: ConcertListItemProps) => (
      <div>
        <span>{data.id}</span>
      </div>
    ),
  }
})

const getFutureConcertsMock = vi.mocked(getFutureConcerts)

describe('FutureConcerts component', () => {
  it('shows all the data', async () => {
    vi.useFakeTimers().setSystemTime(new Date('2020-01-01'))

    getFutureConcertsMock.mockResolvedValue({
      data: [mockConcert],
      error: undefined,
    })

    const ResolvedFutureConcerts = await resolvedComponent<Props>(
      FutureConcerts,
      {
        skip: 0,
        first: 3,
      }
    )

    const { container } = render(<ResolvedFutureConcerts />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('mock-id')).toBeInTheDocument()
  })
})
