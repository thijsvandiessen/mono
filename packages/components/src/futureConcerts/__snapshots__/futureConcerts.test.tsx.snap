// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FutureConcerts component > shows all the data 1`] = `
<div>
  <div>
    <div>
      <span>
        mock-id
      </span>
    </div>
  </div>
</div>
`;
